src/shim/umq/hwq.cpp

```diff
@@ -202,6 +216,10 @@ fill_slot_and_send(volatile host_queue_packet_t *pkt, void *payload, size_t size
   std::memcpy(data, payload, size);
```

Comment:
I believe you need a flush right after memcpy.

Reply:
yes, added the flush here.